File: JavaScript/Math-Array-String/81.js
// 575. Distribute Candies

/**
 * @param {number[]} candyType
 * @return {number}
 */
var distributeCandies = function(candyType) {
    let maxAllowed = candyType.length / 2;
    let uniqueCandies = new Set(candyType).size;
    return Math.min(uniqueCandies, maxAllowed);
};

// Test cases
console.log(distributeCandies([1,1,2,2,3,3])); // Output: 3
console.log(distributeCandies([1,1,2,3]));     // Output: 2
console.log(distributeCandies([6,6,6,6]));     // Output: 1

File: Python/Math-Array-String/81.py
# 575. Distribute Candies

from typing import List

class Solution:
    def distributeCandies(self, candyType: List[int]) -> int:
        return min(len(set(candyType)), len(candyType) // 2)
    

sol = Solution()
output = sol.distributeCandies([1,1,3,2])
print(output)  # Output: 2